azure-deploy-template.yml

```diff
@@ -0,0 +1,66 @@
+parameters:
+  azureSubscription:
+  resourceGroupPrefix:
+  environmentName:
+  slackWebhookUrl:
+
+jobs:
+  - job: run_deploy_script
+    displayName: Run deploy script
+    steps:
+      - task: UseRubyVersion@0
+        displayName: 'Use Ruby >= 2.4'
+      - task: AzureCLI@1
+        displayName: 'Deploy Azure'
+        inputs:
+          azureSubscription: ${{parameters.azureSubscription}}
+          scriptPath: '$(System.DefaultWorkingDirectory)/Build/repository/bin/azure-deploy'
+          arguments: 'test --skip-login --skip-confirmation --docker-tag=$(Build.BuildNumber)'
+          workingDirectory: '$(System.DefaultWorkingDirectory)/Build/repository'
+        env:
+          GIT_COMMIT_HASH: $(Release.Artifacts.Build.SourceVersion)
+  - job: swap_app_service_slots
+    dependsOn: run_deploy_script
+    displayName: Swap app service slots
+    steps:
+      - task: AzureAppServiceManage@0
+        displayName: 'Swap Slots: ${{parameters.resourceGroupPrefix}}-app-as'
+        inputs:
+          azureSubscription: ${{parameters.azureSubscription}}
+          WebAppName: '${{parameters.resourceGroupPrefix}}-app-as'
+          ResourceGroupName: '${{parameters.resourceGroupPrefix}}-app'
+          SourceSlot: staging
+      - task: AzureAppServiceManage@0
+        displayName: 'Swap Slots: ${{parameters.resourceGroupPrefix}}-app-vsp-as'
+        inputs:
+          azureSubscription: 'azdo.pipelines.cip.S118T.arm03ce3ff5-9a61-4525-a063-6ecde34874cf'
+          WebAppName: '${{parameters.resourceGroupPrefix}}-app-vsp-as'
+          ResourceGroupName: '${{parameters.resourceGroupPrefix}}-app'
+          SourceSlot: staging
+  - job: clean_up
+    dependsOn: swap_app_service_slots
+    displayName: Clean up
+    steps:
+      - task: AzureCLI@1
+        displayName: 'Delete migration runner container'
+        inputs:
+          azureSubscription: ${{parameters.azureSubscription}}
+          scriptLocation: inlineScript
+          inlineScript: 'az container delete --name ${{parameters.resourceGroupPrefix}}-app-migration-runner-aci --resource-group s118t01-app -y'
+  - job: send_slack_notification
+    dependsOn: clean_up
+    displayName: Send Slack notification
+    steps:
+      - task: Bash@3
+        displayName: 'Send Slack Notification'
+        inputs:
+          targetType: filePath
+          filePath: './$(System.DefaultWorkingDirectory)/Build/repository/bin/slack-alert'
+          arguments: 'test $(Build.BuildNumber) $(Release.ReleaseId) ${{parameters.slackWebhookUrl}} SUCCESS'
```

Comment:
Do we consider the `slackWebhookUrl` a secret? I'm not sure, I haven't done much with them before. 

If we do we'd probably want to make it a secret variable in the same way we do the Docker password (https://docs.microsoft.com/en-us/azure/devops/pipelines/process/variables?view=azure-devops&tabs=yaml%2Cbatch#secret-variables)

Reply:
Ah, good catch. I think it is _relatively_ sensitive in that people could spam the channel if they had it, so it's best to keep it a secret.

Reply:
Ah I see, cool. The risk is relatively low at the moment because with out current setup only people with access can see the output of the pipeline